### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by natemcmaster**
Remove usages of project.json
Also cleans out unnecessary AssemblyInfo files.

cref https://github.com/aspnet/Universe/issues/469

### CODE REVIEW

File: BasicMiddleware.sln [hunk 1]
```diff
@@ -1,116 +1,112 @@
...
+	ProjectSection(SolutionItems) = preProject
	EndProjectSection
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "test", "test", "{8437B0F3-3894-4828-A945-A9187F37631D}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.AspNetCore.HttpOverrides.Tests", "test\Microsoft.AspNetCore.HttpOverrides.Tests\Microsoft.AspNetCore.HttpOverrides.Tests.csproj", "{D6341B92-3416-4F11-8DF4-CB274296175F}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{99B72A07-32D6-434D-B44D-D064E3C13E08}"
+	ProjectSection(SolutionItems) = preProject
		NuGetPackageVerifier.json = NuGetPackageVerifier.json
```
Comment (BrennanConroy) [inline BasicMiddleware.sln]: What are these? GitHub shows them as `^M`

Comment (natemcmaster) [inline BasicMiddleware.sln]: dotnet-migrate did something funky. Will fix.

**Comment by natemcmaster** [comment]
🆙 :date:

**Comment by BrennanConroy** [comment]
:shipit:

**Pull request merged**